### ISSUE TITLE
ES6 destructuring in object assignment broken?

### ISSUE THREAD

**Issue by callumacrae** [eslint/eslint#3963]
This is, as far as I can see, valid ES6 code:

```js
[].map((letter, index) => ({ ...letter, index }))
```

Babel compiles it fine. ESLint, with the es6 flag enabled, says "Unexpected token ..". Any ideas what is causing this? Is it a bug, or am I doing something wrong?

Thanks!

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by callumacrae** [comment]
turns out my eslint version was massively out of date, oops. fixed now!

**Comment by michaelficarra** [comment]
It's still not ES6. Probably ES2016, though. Object rest/spread is still a proposal.
